### PR TITLE
feat(docs): Document `contrib/anaisdg` packages

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: libflux/flux-core/src/bin/fluxdoc.rs [hunk 1]
```diff
@@ -237,7 +237,7 @@ impl<'a> example::Executor for CLIExecutor<'a> {
 // high standard going forward.
 //
 // See https://github.com/influxdata/flux/issues/4141 for tacking removing of this list.
-const EXCEPTIONS: [&str; 97] = [
+const EXCEPTIONS: [&str; 94] = [
```
Comment (Marwes) [inline libflux/flux-core/src/bin/fluxdoc.rs:240]:
```suggestion
const EXCEPTIONS: &[&str] = &[
```

File: stdlib/contrib/anaisdg/anomalydetection/mad.flux [hunk 2]
```diff
@@ -1,9 +1,14 @@
+// Package anomalydetection detects anomalies in time series data.
 package anomalydetection
 
 
 import "math"
 import "experimental"
 
+// mad TODO.
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/anomalydetection/mad.flux:8]:
```suggestion
// mad uses the median absolute deviation (MAD) algorithm to detect anomalies in a data set.
//
// Input data requires `_time` and `_value` columns.
// Output data is grouped by `_time` and includes the following columns of interest:
//
// - **\_value**: difference between of the original `_value` from the computed MAD
//   divided by the median difference.
// - **MAD**: median absolute deviation of the group.
// - **level**: anomaly indicator set to either `anomaly` or `normal`.
```

File: stdlib/contrib/anaisdg/anomalydetection/mad.flux [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
+//
+// ## Parameters
+// - threshold: deviation at which to report anamolies
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/anomalydetection/mad.flux:11]:
```suggestion
// - threshold: Deviation threshold for anomalies.
// - tables: Input data. Default is piped-forward data (`<-`).
//
// ## Examples
// 
// ### Use the MAD algorithm to detect anomalies
// ```
// import "contrib/anaisdg/anomalydetection"
// import "sampledata"
// 
// < sampledata.float()
// >     |> anomalydetection.mad(threshold: 1.0)
```

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux [hunk 4]
```diff
@@ -1,10 +1,12 @@
+// Package statsmodels provides code for calculating linear regression.
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:1]:
```suggestion
// Package statsmodels provides functions for calculating statistical models.
```

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
 package statsmodels
 
-
 import "math"
 import "generate"
 
-// performs linear regression, calculates y_hat, and residuals squared (rse) 
+// linearRegression performs linear regression. It calculates and returns
+// [*ŷ*](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
+// and [residual sum of errors](https://en.wikipedia.org/wiki/Residual_sum_of_squares) (`rse`).
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:9]:
~~~suggestion
// linearRegression performs a linear regression.
//
// It calculates and returns [*ŷ*](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
// and [residual sum of errors](https://en.wikipedia.org/wiki/Residual_sum_of_squares) (`rse`).
// Output data includes the following columns:
//
// - **N**: number of points in the calculation.
// - **slope**: slope of the calculated regression.
// - **sx**: ...
// - **sxx**: ...
// - **sxy**: ...
// - **sy**: ...
// - **errors**: ...
// - **x**: ...
// - **y**: ...
// - **y\_hat**: ...
//
// ## Parameters
// - tables: Input data. Default is piped-forward data (`<-`).
//
// ## Examples
//
// ### Perform a linear regression on a dataset
// ```
// import "contrib/anaisdg/statsmodels"
// import "sampledata"
//
// < sampledata.float()
// >     |> statsmodels.linearRegression()
// ```
~~~

@Anaisdg, any chance you could shed some light on what these columns represent?

**Review by sanderson — CHANGES_REQUESTED** [review]

**Comment by sanderson** [comment]
@pierwill When you modify any of the `.flux` files in `stdlib`, you have to regenerate libflux for tests to pass. From the root of the repo, run:

```
$ make generate
$ go generate ./stdlib ./libflux/go/libflux
```

**Comment by pierwill** [comment]
> @pierwill When you modify any of the `.flux` files in `stdlib`, you have to regenerate libflux for tests to pass. From the root of the repo, run:
> 
> ```
> $ make generate
> $ go generate ./stdlib ./libflux/go/libflux
> ```

I have this as a git hook. I'll check to see it's working...

File: libflux/flux-core/src/bin/fluxdoc.rs (hunk 1 above)
Comment (pierwill) [reply libflux/flux-core/src/bin/fluxdoc.rs]: Thank you 🙏

Comment (pierwill) [reply libflux/flux-core/src/bin/fluxdoc.rs]: @sanderson No need to update array length after this.

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux (hunk 5 above)
Comment (Anaisdg) [reply stdlib/contrib/anaisdg/statsmodels/linearreg.flux:10]: sx = sum of x 
sxx = sum of x squared
sxy = sum of x*y
sy = sum of y
y = field value
y_hat = linear regression values
errors = residual sum of squares defined by (r.y - r.y_hat) ^ 2.0 in this context

**Review by Anaisdg — APPROVED** [review]

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux (hunk 5 above)
Comment (pierwill) [reply stdlib/contrib/anaisdg/statsmodels/linearreg.flux]: @Anaisdg: how would you describe `x` here? (If `y` is field value).

Comment (Anaisdg) [reply stdlib/contrib/anaisdg/statsmodels/linearreg.flux]: @ x is an index [1,2,3,4...n] with the assumption that the timestamps are regularly spaced.

File: libflux/flux-core/src/bin/fluxdoc.rs [hunk 6]
```diff
@@ -251,13 +251,11 @@ impl<'a> example::Executor for CLIExecutor<'a> {
 //
 // See https://github.com/influxdata/flux/issues/4141 for tacking removing of this list.
 const EXCEPTIONS: &[&str] = &[
+    "array",
```
Comment (sanderson) [inline libflux/flux-core/src/bin/fluxdoc.rs:254]: Looks like this branch needs to be rebased to master.

File: stdlib/contrib/anaisdg/anomalydetection/mad.flux [hunk 7]
```diff
@@ -1,9 +1,32 @@
+// Package anomalydetection detects anomalies in time series data.
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/anomalydetection/mad.flux:1]:
```suggestion
// Package anomalydetection detects anomalies in time series data.
// 
// introduced: 0.90.0
// tags: anomaly detection
//
```

File: stdlib/contrib/anaisdg/anomalydetection/mad.flux [hunk 8]
```diff
@@ -1,9 +1,32 @@
...
+//
+// ## Examples
+// 
+// ### Use the MAD algorithm to detect anomalies
+// ```
+// import "contrib/anaisdg/anomalydetection"
+// import "sampledata"
+// 
+// < sampledata.float()
+// >     |> anomalydetection.mad(threshold: 1.0)
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/anomalydetection/mad.flux:29]:
~~~suggestion
// >     |> anomalydetection.mad(threshold: 1.0)
// ```
//
// tags: transformations
//
~~~

File: stdlib/contrib/anaisdg/anomalydetection/mad.flux [hunk 9]
```diff
@@ -1,9 +1,32 @@
...
+// Input data requires `_time` and `_value` columns.
+// Output data is grouped by `_time` and includes the following columns of interest:
+//
+// - **\_value**: difference between of the original `_value` from the computed MAD
+//   divided by the median difference.
+// - **MAD**: median absolute deviation of the group.
+// - **level**: anomaly indicator set to either `anomaly` or `normal`.
+//
+// ## Parameters
+// - threshold: Deviation threshold for anomalies.
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/anomalydetection/mad.flux:19]: There is a fix that lets up add input data parameters `tables`/`table` to the docs now. When you rebase, this should pass lint now.

```suggestion
// - threshold: Deviation threshold for anomalies.
// - table: Input data. Default is piped-forward data (`<-`).
```

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux [hunk 10]
```diff
@@ -1,10 +1,38 @@
+// Package statsmodels provides functions for calculating statistical models.
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:1]:
```suggestion
// Package statsmodels provides functions for calculating statistical models.
//
// introduced: 0.90.0
//
```

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux [hunk 11]
```diff
@@ -1,10 +1,38 @@
...
 package statsmodels
 
 
 import "math"
 import "generate"
 
-// performs linear regression, calculates y_hat, and residuals squared (rse) 
+// linearRegression performs a linear regression.
+//
+// It calculates and returns [*ŷ*](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:10]:
```suggestion
// It calculates and returns [**ŷ**](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
```

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux [hunk 12]
```diff
@@ -1,10 +1,38 @@
...
+// - **sx**: Sum of x.
+// - **sxx**: Sum of x squared.
+// - **sxy**: Sum of x*y.
+// - **sy**: Sum of y.
+// - **errors**: Residual sum of squares.
+//   Defined by `(r.y - r.y_hat) ^ 2` in this context
+// - **x**: An index [1,2,3,4...n], with the assumption that the timestamps are regularly spaced.
+// - **y**: Field value
+// - **y\_hat**: Linear regression values
+//
```
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:25]:
```suggestion
//
// ## Parameters
// - tables: Input data. Default is piped-forward data (`<-`).
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by sanderson — APPROVED** [review]

**Pull request merged**